packages/input/src/input.tsx

```diff
@@ -47,15 +51,22 @@ export interface InputProps
  * Element that allows users enter single valued data.
  */
 export const Input = forwardRef<InputProps, "input">((props, ref) => {
-  const styles = useMultiStyleConfig("Input", props)
-  const ownProps = omitThemingProps(props)
+  const { htmlSize, ...rest } = props
+
+  const styles = useMultiStyleConfig("Input", rest)
+  const ownProps = omitThemingProps(rest)
   const input = useFormControl<HTMLInputElement>(ownProps)
   const _className = cx("chakra-input", props.className)
 
+  // This is used to remove the default width=100% when size is set, which otherwise conflicts
+  const { width, ...fieldStyles } = styles.field
+  const cssStyles = htmlSize ? { ...fieldStyles } : { ...styles.field }
```

Comment:
I'm not quite sure we want to predict the user's goal here.

From the perspective of the using side I find it much easier to understand whats happening when the width is explicitly set like `<Input htmlSize="10" width="auto" />`.

Otherwise we would need to filter the props `width` and `w` respectively.

What do you think about that?

Reply:
I think that building around the default theme width is overcomplicating the component. I'd be happy to remove it to make the component more transparent in usage and less funky. Just didn't wanted to make too many changes everywhere.
Removing the default theme width might result in a breaking change tho, as we don't know how many people rely on this, as it's a default. 😕

Reply:
> I'd be happy to remove it to make the component more transparent in usage and less funky.

Sounds great! 

> Removing the default theme width might result in a breaking change 

Yes, this would be a breaking change IMO and not necessary to introduce a new html prop.

Reply:
Actually I don't get why we wouldn't need the new htmlProp ? We still have the issue that theme size and native size conflict here ?

Reply:
I think I did not express my self clearly 😅 
It is great to add support for the html `size` attribute! But I think this does not mean to introduce a breaking change to the theme. 
If a user wants to use `htmlSize` I think it is much cleaner to let them use `<Input htmlSize="10" width="auto" />` instead of internally omitting the styleProp `width`.

Reply:
Ahhh I right I got it. Docs thing then for the usage of the htmlSize and to override default width with auto. Gotcha. Will make those changes and open a PR in docs aswell 😄